Assets/Scripts/HoldObject.cs

```diff
@@ -5,34 +5,33 @@
 public class HoldObject : MonoBehaviour
 {
     private GameObject pickedUpObject;
+    private GameObject targetedObject;
     private Quaternion pickedObjectRotation;
     private float floorHeight = 0.5f;
 
     public float interactionDistance = 10f;
 
     void Update()
     {
+        targetedObject = HighlightPickupable();
         if (Input.GetKeyDown("e"))
         {
             if (pickedUpObject == null)
-                PickUp();
+            {
+                if (targetedObject != null)
```

Comment:
```suggestion
                if (pickedUpObject == null && targetedObject != null)
```
Please combine them :D

Reply:
oh sorry that's actually needed